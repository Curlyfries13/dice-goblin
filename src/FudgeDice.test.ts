import seedrandom from 'seedrandom';

import FudgeDiceGroup from 'FudgeDice';

it('should initialize a single fudge die by default', () => {
  const dice = new FudgeDiceGroup();
  expect(dice.sides.value()).toBe(3);
  expect(dice.count.value()).toBe(1);
});

it('should behave predictably with provided seed', () => {
  const rng = seedrandom('test');
  const dice = new FudgeDiceGroup(3, rng);
  expect(dice.roll()).toMatchSnapshot();
});
