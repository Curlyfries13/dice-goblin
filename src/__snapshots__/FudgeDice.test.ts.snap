// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should behave predictably with provided seed 1`] = `2`;
